### PR TITLE
Added dislikes, managed interactions with likes

### PR DESCRIPTION
**Pull request by alessandrobelfiore**

### CODE REVIEW

File: monolith/views/stories.py [hunk 1]
```diff
@@ -32,21 +32,74 @@ def _random_story(message=''):
...
         message = 'You\'ve already liked this story!'
-    return _stories(message)
+    return _story(story_id, message)
+
+@stories.route('/story/<story_id>/dislike')
+@login_required
+def _dislike(story_id):
+    q = Dislike.query.filter_by(disliker_id=current_user.id, story_id=story_id)
+    story = Story.query.filter_by(id=story_id).first()
+    if q.first() == None:
```
Comment (sduoase) [inline monolith/views/stories.py]: Same as https://github.com/sduoase/STAR_DICES/pull/53/files#diff-7b60d0b7d71eb2e20a3bed333226ad19R42.

File: monolith/views/stories.py [hunk 2]
```diff
@@ -32,21 +32,74 @@ def _random_story(message=''):
...
+        message = 'Dislike added!'
+    else:
+        message = 'You\'ve already disliked this story!'
+    return _story(story_id, message)
+
+@stories.route('/story/<story_id>/remove_like')
+@login_required
+def _remove_like(story_id):
+    l = Like.query.filter_by(liker_id=current_user.id, story_id=story_id).first()
+    if l == None:
```
Comment (sduoase) [inline monolith/views/stories.py]: Same as https://github.com/sduoase/STAR_DICES/pull/53/files#diff-7b60d0b7d71eb2e20a3bed333226ad19R42.

File: monolith/views/stories.py [hunk 3]
```diff
@@ -32,21 +32,74 @@ def _random_story(message=''):
...
+        db.session.commit()
+        message = 'You removed your like'
+    return _story(story_id, message)
+    
+    
+@stories.route('/story/<story_id>/remove_dislike')
+@login_required
+def _remove_dislike(story_id):
+    d = Dislike.query.filter_by(disliker_id=current_user.id, story_id=story_id).first()
+    if d == None:
```
Comment (sduoase) [inline monolith/views/stories.py]: Same as https://github.com/sduoase/STAR_DICES/pull/53/files#diff-7b60d0b7d71eb2e20a3bed333226ad19R42.

File: monolith/include/python3.6m [hunk 4]
```diff
@@ -0,0 +1 @@
+/usr/include/python3.6m
```
[the remark this answers is not in the document]
Comment (alessandrobelfiore) [reply monolith/include/python3.6m]: I don't know how that got there

File: monolith/database.py [hunk 5]
```diff
@@ -70,6 +71,20 @@ class Like(db.Model):
     liked_id = db.Column(db.Integer, db.ForeignKey('user.id')) # TODO: duplicated ?
     liker = relationship('User', foreign_keys='Like.liker_id')
 
+    marked = db.Column(db.Boolean, default = False) # True iff it has been counted in Story.likes
+    
+class Dislike(db.Model):
+    __tablename__ = 'dislike'
+    
+    disliker_id = db.Column(db.Integer, db.ForeignKey('user.id'), primary_key=True)
+    liker = relationship('User', foreign_keys='Dislike.disliker_id')
```
[the remark this answers is not in the document]
Comment (alessandrobelfiore) [reply monolith/database.py]: As the liker, the person who clicked dislike

[the remark this answers is not in the document]
Comment (sduoase) [reply monolith/database.py]: I meant that on line 80 the variable should probably be called disliker?

File: monolith/database.py [hunk 6]
```diff
@@ -70,6 +71,20 @@ class Like(db.Model):
...
+
+    story_id = db.Column(db.Integer, db.ForeignKey('story.id'), primary_key=True)
+    author = relationship('Story', foreign_keys='Dislike.story_id')
+
+    disliked_id = db.Column(db.Integer, db.ForeignKey('user.id')) # TODO: duplicated ?
+    disliker = relationship('User', foreign_keys='Dislike.disliker_id')
```
[the remark this answers is not in the document]
Comment (sduoase) [reply monolith/database.py]: Here as well I'm referring to the variable on line 86.

File: monolith/database.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (alessandrobelfiore) [reply monolith/database.py]: Ah definitely ahah

File: monolith/database.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (alessandrobelfiore) [reply monolith/database.py]: Right and there's also a typo in the original Like class, with the same error

**Comment by sduoase** [comment]
Ping for review.

**Pull request merged**